### PR TITLE
Preload images for Infinity Play Lab

### PR DESCRIPTION
**Pull request by breville**
Preload images for avatars, backgrounds, and projectiles for Infinity.
- Ensured that image preloading continues working at /s/playlab which uses original Studio skin.
- Most dramatic change is making ProjectileClassNames per-skin.
- Continued current implementation of loading background images from skin.backgroundChoicesK1.

### CODE REVIEW

**Comment by breville** [comment]
@Bjvanminnen CR please? :)

**Comment by biarm** [comment]
Hi Brendan, I am exploring the possibilities to use Blockly/code.org in our online quest multi-school project. Play lab seems to be an excellent app for us, but we would need a couple more images and skins for the stories we are planning to make. Is it possible by now to upload custom images into the playlab? Is it going to be possible in the nearest future?

File: apps/src/studio/skins.js [hunk 1]
```diff
@@ -16,11 +16,24 @@ var VISIBLE_VALUE = constants.VISIBLE_VALUE;
...
 
 function loadInfinity(skin, assetUrl) {
-  skin.preloadAssets = false;
+  skin.preloadAssets = true;
 
   skin.defaultBackground = 'leafy';
   skin.projectileFrames = 10;
 
+  // NOTE: all class names should be unique.  eventhandler naming won't work
+  // if we name a projectile class 'left' for example.
```
Comment (Bjvanminnen) [inline apps/src/studio/skins.js]: Think this is important to validate this somewhere?

**Comment by Bjvanminnen** [comment]
lgtm

File: apps/src/studio/skins.js (hunk 1 above)
Comment (breville) [inline apps/src/studio/skins.js]: I think our naming conventions for these classes are so different that we should be okay for now...

**Comment by breville** [comment]
@biarm you can see from recent commits that we now have two separate skins for play lab, the legacy "studio" one and the new "infinity" one.  you could conceivably add a new skin with your own backgrounds and sprites (which can now include walking cycles), using the "infinity" skin as a guide.

**Pull request merged**